packages/valory/skills/twitter_scoring_abci/behaviours.py

```diff
@@ -123,38 +124,96 @@ class TwitterCollectionBehaviour(TwitterScoringBaseBehaviour):
 
-    def async_act(self) -> Generator:
-        """Do the act, supporting asynchronous execution."""
+    def _check_daily_limit(self) -> Tuple:
+        """Check if the daily limit has exceeded or not"""
+        try:
+            number_of_tweets_pulled_today = int(
+                self.synchronized_data.ceramic_db["module_data"]["twitter"][
+                    "number_of_tweets_pulled_today"
+                ]
+            )
+            last_tweet_pull_window_reset = float(
+                self.synchronized_data.ceramic_db["module_data"]["twitter"][
+                    "last_tweet_pull_window_reset"
+                ]
+            )
+        except KeyError:
+            number_of_tweets_pulled_today = 0
+            last_tweet_pull_window_reset = cast(
+                SharedState, self.context.state
+            ).round_sequence.last_round_transition_timestamp.timestamp()
 
-        with self.context.benchmark_tool.measure(self.behaviour_id).local():
+        current_time = cast(
+            SharedState, self.context.state
+        ).round_sequence.last_round_transition_timestamp.timestamp()
 
-            # Get mentions from Twitter
-            tweets, latest_mention_tweet_id = yield from self._get_twitter_mentions()
+        # Window has expired
+        if current_time >= last_tweet_pull_window_reset + ONE_DAY:
+            return False, 0, current_time
 
-            # Get hashtags from Twitter
-            if tweets != TwitterCollectionRound.ERROR_PAYLOAD:
-                (
-                    tweets,
-                    latest_hashtag_tweet_id,
-                ) = yield from self._get_twitter_hashtag_search(tweets)
+        # Reached max number of tweets
+        if number_of_tweets_pulled_today >= self.params.max_tweet_pulls_allowed:
+            return True, number_of_tweets_pulled_today, last_tweet_pull_window_reset
 
-                # Keep the max latest_tweet_id
-                valid_latest_ids = [
-                    i for i in (latest_mention_tweet_id, latest_hashtag_tweet_id) if i
-                ]
-                if valid_latest_ids:
-                    latest_mention_tweet_id = max(valid_latest_ids)
+        # Window has not expired and we have not reached the max number of tweets
+        return False, number_of_tweets_pulled_today, last_tweet_pull_window_reset
+
+    def async_act(self) -> Generator:
+        """Do the act, supporting asynchronous execution."""
+
+        with self.context.benchmark_tool.measure(self.behaviour_id).local():
 
-            if tweets == TwitterCollectionRound.ERROR_PAYLOAD:
-                payload_data = TwitterCollectionRound.ERROR_PAYLOAD
+            (
+                has_limit_reached,
+                number_of_tweets_pulled_today,
+                last_tweet_pull_window_reset,
+            ) = self._check_daily_limit()
+            if has_limit_reached:
+                payload = TwitterCollectionRound.ERROR_PAYLOAD
```

Comment:
The payload is being overwritten after this. I think this should be either payload_data (and we check payload.content on the round side), or we take care of this being overriden.